tests/AppBundle/Controller/JobControllerTest.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Tests\AppBundle\Controller;
+
+use Doctrine\ORM\Tools\SchemaTool;
+use Liip\FunctionalTestBundle\Test\WebTestCase;
+
+class JobControllerTest extends WebTestCase
+{
+    public function setUp()
+    {
+        $em = $this->getContainer()->get('doctrine')->getManager();
+        if (!isset($metadatas)) {
+            $metadatas = $em->getMetadataFactory()->getAllMetadata();
+        }
+        $schemaTool = new SchemaTool($em);
+        $schemaTool->dropDatabase();
+        if (!empty($metadatas)) {
+            $schemaTool->createSchema($metadatas);
+        }
+        $this->postFixtureSetup();
+
+        $fixtures = [
+            \AppBundle\DataFixtures\CategoryFixture::class,
+            \AppBundle\DataFixtures\JobFixture::class
+        ];
+        $this->loadFixtures($fixtures);
+    }
+
+    /**
+     * Test job index page and see the job.
+     */
+    public function testIndex()
+    {
+        $client = static::createClient();
+        $max_jobs_on_homepage = static::$kernel->getContainer()->getParameter('max_jobs_on_homepage');
+
+        $crawler = $client->request('GET', '/');
+
+        $this->assertCount(0, $crawler->filter('.jobs td.position:contains("Expired")'));
+
+        $this->assertTrue($crawler->filter('.category_programming tr')->count() <= $max_jobs_on_homepage);
+        $this->assertCount(0, $crawler->filter('.category_design .more_jobs'));
+        $this->assertCount(1, $crawler->filter('.category_programming .more_jobs'));
+
+        $job = $this->getMostRecentProgrammingJob();
+        $this->assertCount(1, $crawler->filter('.category_programming tr')->first()->filter(sprintf('a[href*="/%d/"]', $job->getId())));
+
+        $link = $crawler->selectLink('Web Developer')->first()->link();
+        $crawler = $client->click($link);
+        $this->assertEquals('AppBundle\Controller\JobController::showAction', $client->getRequest()->attributes->get('_controller'));
+        $this->assertEquals($job->getCompanySlug(), $client->getRequest()->attributes->get('company'));
+        $this->assertEquals($job->getLocationSlug(), $client->getRequest()->attributes->get('location'));
+        $this->assertEquals($job->getPositionSlug(), $client->getRequest()->attributes->get('position'));
+        $this->assertEquals($job->getId(), $client->getRequest()->attributes->get('id'));
+    }
+
+    /**
+     * Test nonexisting routes to job.
+     */
+    public function testShow()
```

Comment:
separate this tests in two tests: testShowWithNotExistentJob and testShowWithEspiredJob
Better to write test with one test case